Sources/NIOHTTPClient/SwiftNIOHTTP.swift

```diff
@@ -128,6 +131,30 @@ public class HTTPClient {
 
         let task = Task(future: promise.futureResult)
 
+        let connection = pool.getConnection(for: request)
+        
+        connection.flatMap { connection -> EventLoopFuture<()> in
+            let channel = connection.channel
+            let addedFuture: EventLoopFuture<Void>
+            if let readTimeout = timeout.read {
+                addedFuture = channel.pipeline.addHandler(IdleStateHandler(readTimeout: readTimeout))
+            } else {
+                addedFuture = channel.eventLoop.makeSucceededFuture(())
+            }
+            return addedFuture.flatMap {
+                let taskHandler = TaskHandler(task: task, delegate: delegate, promise: promise, redirectHandler: redirectHandler)
+                task.setChannel(channel)
+                return channel.pipeline.addHandler(taskHandler)
+            }.flatMap {
+                return channel.writeAndFlush(request)
+            }
+            }.whenFailure { error in
+                promise.fail(error)
+        }
+        
+        return task
+        
+        /*
```

Comment:
Is the commented out code here for a reason?

Reply:
Nope, has just kept it here for reference. Will be removed.